core/src/main/scala/java/text/DecimalFormat.scala

```diff
@@ -1,57 +1,521 @@
+        unscaled.divide(
+          JavaBigDecimal.TEN.pow(n.precision - newPrecision),
+          if (isJustFraction) RoundingMode.HALF_UP else getRoundingMode
+        )
+      }
+
+      (
+        scalePrecision,
+        originalDecimalPosition - newIntegerSize
+      )
+    }
+
+  // Based upon number count, add in the number of group separators
+  private def totalDigitsWritten(count: Int): Int = {
+    if (isGroupingUsed && getGroupingSize > 0 && count > 0) {
+      count + (count / getGroupingSize)
+    } else count
+  }
+
+  private def repeatDigits(count: Int, c: Char): String = (0 until count).map{ _ => c }.mkString
+
+  // Handle formatting any big decimal...I'm sure this algorithm can be optimized
+  // ...Trying to get a mostly correct/easier to read/understand implementation first
+  // TODO: Currently ignoring FieldPosition argument
+  private def subFormat(number: JavaBigDecimal, toAppendTo: StringBuffer, pos: FieldPosition): StringBuffer = {
+    val isNegative: Boolean = number.signum == -1
+
+    // Add Prefix
+    val prefix: String = if (isNegative) getNegativePrefix() else getPositivePrefix()
+    toAppendTo.append(prefix)
+
+    val multiplied: JavaBigDecimal = number.multiply(JavaBigDecimal.valueOf(getMultiplier)).abs
+
+    // Round the target number based upon expected fractions, so we can compare it to the integer
+    val (targetNumber: JavaBigDecimal, expPower: Int) =
+      if (useScientificNotation) {
+        getExponentNumberAndPower(multiplied)
+      } else {
+        (multiplied.setScale(max(getMaximumFractionDigits(), getMinimumFractionDigits()), getRoundingMode), 0)
+      }
+
+    val integerStrBuilder = new StringBuilder()
+    val integerPart: JavaBigDecimal = new JavaBigDecimal(targetNumber.toBigInteger, 0)
+    var integerDigitsWritten: Int = 0
 
-  override final def format(obj: AnyRef, toAppendTo: StringBuffer, pos: FieldPosition): StringBuffer = ???
+    if ((integerPart.compareTo(JavaBigDecimal.ZERO) == 0) || (getMaximumIntegerDigits == 0)) {
+      integerStrBuilder.append(symbols.getZeroDigit)
+      integerDigitsWritten += 1
+    } else {
+      integerDigitsWritten += formatNumber(integerPart.toBigInteger, integerStrBuilder, true)
+    }
 
-  override final def parseObject(source: String, pos: ParsePosition): AnyRef = ???
+    // Add integer digit padding if needed
+    if (integerDigitsWritten < getMinimumIntegerDigits) {
+      (integerDigitsWritten until getMinimumIntegerDigits).foreach{ _ =>
+        handleGroupSeparator(integerStrBuilder, integerDigitsWritten)
+
+        integerStrBuilder.append(symbols.getZeroDigit)
+        integerDigitsWritten += 1
+      }
+    }
+
+    // We have the integer portion ready, append it to the builder
+    toAppendTo.append(integerStrBuilder.result.reverse)
+
+    // Compare the target number to the strictly-integer value...
+    targetNumber.compareTo(integerPart) match {
+      // We have a fractional value...
+      case 1  =>
```

Comment:
Yeah this whole pattern probably should be cleaned up a little bit, I hate dealing with "compareTo"